pkg/operator/sync_kubecontrollermanager_v311_00.go

```diff
@@ -90,6 +121,28 @@ func syncKubeControllerManager_v311_00_to_latest(c KubeControllerManagerOperator
 	return resourcemerge.ApplyGenerationAvailability(versionAvailability, actualDeployment, errors...), errors
 }
 
+func applyKubeControllerManagerOperatorConfig(client operatorsv1alpha1client.KubeControllerManagerOperatorConfigsGetter, required *v1alpha1.KubeControllerManagerOperatorConfig) (*v1alpha1.KubeControllerManagerOperatorConfig, bool, error) {
```

Comment:
this function needs to be weird.  Only adjust the imagepullspec if it mismatches or create if it is missing entirely.